src/modules/connections/connection-manager.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import {ConnectionsRepositoryService} from './connections-repository.service';
+import { Connection, ConnectionOptions, createConnection } from 'typeorm';
+import {ConnectionData} from './connections.interfaces';
+
+@Injectable()
+export class ConnectionManagerService {
+  private connections: Map<number, Connection> = new Map();
+  constructor(
+    private connectionRepository: ConnectionsRepositoryService,
+  ) {}
+
+  async getConnection(id: number): Promise<Connection> {
+    let connection = this.connections.get(id);
+    try {
+      if (connection && connection.isConnected) {
+        return connection;
+      } else if (connection) {
+        return await connection.connect();
+      }
+      const connectionData: ConnectionData = await this.connectionRepository.getDataForConnectionCreating(id);
+      if (!connectionData) {
+        return null;
```

Comment:
Will the user (frontend) know, that some error has occurred, like information not found?

Reply:
Obviously - no, he will get message like "Can't connect to database, please check your data"

Reply:
Ok )